src/wtmp.c

```diff
@@ -40,7 +40,7 @@ int log_wtmp(const char *line, const char *name, const char *host,
     !(defined(LINUX) || defined(__hpux) || defined (_AIX))
   /* This "auxilliary" utmp doesn't exist under linux. */
 
-#if defined(__sparcv9) && !defined(__NetBSD__) && !defined(__FreeBSD__)
+#if defined(__sun)
```

Comment:
Yes, it would.
But I still think using just `defined(__sun)` is better as `struct futmpx` is only used on Solaris, and this should limit us exactly to Solaris.